### ISSUE TITLE
[Bug]: ComboBox: onChange is called twice when clearing the input field

### ISSUE THREAD

**Issue by pedesen** [carbon-design-system/carbon#18137]
### Package

@carbon/react

### Browser

Firefox, Chrome

### Package version

1.70.0

### React version

18.2.0

### Description

There is another issue when ComboBox is used as a fully controlled input (meaning with selectedItem and onChange props provided):

When clicking on the "clear selected item" button, the onChange is called twice: first time with null and second time with the item that was cleared before. This leads to unexpected behavior.

This currently prevents us to upgrade `@carbon/react`, because I was not able to find a workaround for this.

### Reproduction/example

https://stackblitz.com/edit/github-w7mkre-dtgf2u?file=src%2FApp.jsx

### Steps to reproduce

1. Clear combobox
2. Select Item 1
3. Clear combobox again
4. Observe console logs

Problem: After step 3 onChange is called twice: first time with null and second time with the item that was cleared before. This leads to unexpected behavior.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by k02pradeep** [comment]
+1

**Comment by tay1orjones** [comment]
Fixed via https://github.com/carbon-design-system/carbon/pull/18041
